core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -572,6 +572,8 @@ final class ZManaged[-R, +E, +A] private (reservation: ZIO[R, E, Reservation[R,
 
   /**
    * Preallocates the managed resource, resulting in a ZManaged that reserves and acquires immediately and cannot fail.
+   * You should take care that you are not interrupted between running preallocate and actually acquiring the resource
+   * as you might leak otherwise.
    */
   final def preallocate: ZIO[R, E, Managed[Nothing, A]] =
```

Comment:
Do we need to expose this as part of the API? It would be safer if we could only preallocate resources in the context of `preallocateManaged` so we couldn't leak them, or at least it would be much more difficult to do so. But maybe that doesn't give you the power to do what you want with this.

Reply:
I can see your point. To use this safely you would almost always need to use uninterruptible / ensuring which is not something we would want to expose to users. On the other hand this will be often more convenient than preallocateManaged.

Maybe lets wait for a few more opinions and then decide what to do with it.

Reply:
I'd love to keep `preallocate` around as it's a pretty handy combinator for tricky situations.